src/opnsense/mvc/app/models/OPNsense/Kea/KeaDhcpv4.xml

```diff
@@ -36,6 +36,9 @@
                     <AddressFamily>ipv4</AddressFamily>
                     <Required>Y</Required>
                 </subnet>
+                <next_server type="TextField">
```

Comment:
move inside `option_data` section.  When next server is a hostname (or address), the `HostnameField` is likely a better type or when only addresses are allowed `NetworkField`.

Reply:
I've just updated the type to be a NetworkField.
As for the "option_data". Kea does not seem to accept the ``next_server`` field ?
```
 ERROR [kea-dhcp4.dhcp4.0x833e8f000] DHCP4_PARSER_FAIL failed to create or run parser for configuration element subnet4: definition for the option 'dhcp4.next-server' does not exist (/usr/local/etc/kea/kea-dhcp4.conf:142:33)
 ```

Reply:
when inside the option_data block the underscores should be replaced automatically https://github.com/opnsense/core/blob/1e948bfc2f7c69d064d19af5e4e376b143443baa/src/opnsense/mvc/app/models/OPNsense/Kea/KeaDhcpv4.php#L150

Reply:
@h3krn totally my mistake, I was expecting `next-server` to be in the option data block. So the current state of your code looks good.

Reply:
No worries. I was trying to figure out for myself why Kea need to have it this way as the exmaples were not 100% clear. 

But I think I figured it out. The ``next-server`` config does not have a dhcp option number. (Like ``tftp-server-name`` which is dhcp option 66.) It is also absent in https://github.com/isc-projects/kea/blob/master/doc/examples/kea4/all-options.json

 Instead it's defined as ``siaddr``.  If finally found an example here where it's also defined outside of the ``option-data`` block: https://github.com/isc-projects/kea/blob/master/doc/examples/kea4/all-keys.json#L573

Reply:
@h3krn it's easy to get lost in the parameters, this looks good, I've merged it to master. Thanks!